src/promptflow/promptflow/executor/_line_execution_process_pool.py

```diff
@@ -207,65 +180,121 @@ def __exit__(self, exc_type, exc_val, exc_tb):
             self._pool.close()
             self._pool.join()
 
-    def _timeout_process_wrapper(self, run_start_time: datetime, task_queue: Queue, timeout_time, result_list):
-        healthy_ensured_process = HealthyEnsuredProcess(self._executor_creation_func, self.context)
-        healthy_ensured_process.start_new(task_queue)
-
-        if not healthy_ensured_process.process.is_alive():
-            return
-
+    def _process_task(
+        self,
+        process_info,
+        task_queue,
+        timeout_time,
+        result_list
+    ):
         while True:
+            index, pid, process_name, input_queue, output_queue = process_info
```

Comment:
It is not a good practice to use a tuple to carry such a lot information, we may use dataclass or something to carry these.

Reply:
fixed